pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -675,6 +675,72 @@ func TestPullSourcesFromReference(t *testing.T) {
 	assert.Equal(t, 1, len(pullSources))
 }
 
+func TestPullSourcesMirrorFromReference(t *testing.T) {
+	sys := &types.SystemContext{
+		SystemRegistriesConfPath:    "testdata/pull-sources-mirror-reference.conf",
+		SystemRegistriesConfDirPath: "testdata/this-does-not-exist",
+	}
+	registries, err := GetRegistries(sys)
+	assert.Nil(t, err)
+	assert.Equal(t, 4, len(registries))
+
+	for _, tc := range []struct {
+		registry      string
+		digestRef     string
+		digestSources int
+		tagRef        string
+		tagSources    int
+	}{
+		// Registry A has mirrors allow any kind of pull
+		{
+			"registry-a.com/foo/image:latest",
+			"registry-a.com/foo/image@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+			3,
+			"registry-a.com/foo/image:aaa",
+			3,
+		},
+		// Registry B has mirrors allow digests pull only
+		{
+			"registry-b.com/foo/image:latest",
+			"registry-b.com/foo/image@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+			3,
+			"registry-b.com/foo/image:aaa",
+			1,
+		},
+		// Registry C has a mirror allows digest pull only and a mirror allows any kind of pull
+		{
+			"registry-c.com/foo/image:latest",
+			"registry-c.com/foo/image@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+			3,
+			"registry-c.com/foo/image:aaa",
+			2,
+		},
+		// Registry D set mirror-by-digest-only, honor this configuration for registry
+		// Regsitry D has no digest-only set for mirrors table
+		{
+			"registry-d.com/foo/image:latest",
+			"registry-d.com/foo/image@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+			3,
+			"registry-d.com/foo/image:aaa",
+			1,
+		},
+	} {
+		registry, err := FindRegistry(sys, tc.registry)
+		assert.Nil(t, err)
+		assert.NotNil(t, registry)
+		// Digest
+		referenceDigest := toNamedRef(t, tc.digestRef)
+		pullSource, err := registry.PullSourcesFromReference(referenceDigest)
+		assert.Nil(t, err)
+		assert.Equal(t, tc.digestSources, len(pullSource))
+		// Tag
+		referenceTag := toNamedRef(t, tc.tagRef)
+		pullSource, err = registry.PullSourcesFromReference(referenceTag)
+		assert.Nil(t, err)
+		assert.Equal(t, tc.tagSources, len(pullSource))
```

Comment:
```go
digestedName := tc.registry + "@sha256:…"
registry, err := FindRegistry(sys, digestedRef)
…
digestedRef := toNamedRef(t, digestedName)
pullSources, err := …
```
and similarly test behavior of a tagged reference; no need to repeat the digest and tag values in the test case array, no need for a separate `tc.registry` with an ignored tag.

Reply:
Actually, to be even closer to real-world callers,
```go
digestedRef := toNamedRef(t, tc.registry + "@sha256:…")
registry, err := FindRegistry(sys, digestedRef.Name())
```